paperqa/agents/search.py

```diff
@@ -674,6 +699,7 @@ async def get_directory_index(  # noqa: PLR0912
     )
     with progress_bar:
         async with anyio.create_task_group() as tg:
+            processed_counter = Counter()
```

Comment:
```suggestion
            processed_counter: Counter[str] = Counter()
```

When `mypy` makes these requests, it's because `mypy` detects there's a risk of a type error, so it wants to be sure